### PR TITLE
Main

### PR DESCRIPTION
**Pull request by Ch3fUlrich**
Added handling of the case when the number of k-neighbors is bigger than the number of cloud_all points.

### CODE REVIEW

**Comment by JulioEI** [comment]
Hi Sergej! 

Nice on catching the outlier case in which one ask for more neighbors than the sum of points in cloud1 and cloud2!
I believe it wouldn be faster to implement it as a case:
L210
`    cloud_all = np.vstack((cloud1, cloud2)).astype('float32')`
`    idx_sep = cloud1.shape[0]`
`    #Create cloud label`
`    cloud_label = np.hstack((np.ones(cloud1.shape[0]), np.ones(cloud2.shape[0])*2))`
`    n_points = cloud_all.shape[0]`
`    if k>n_points -1: #  all points are neighbors of each other`
`        overlap_1_2 = cloud1.shape[0]*cloud2.shape[0]/(cloud1.shape[0]*(n_points-1))`
`        overlap_2_1 = cloud1.shape[0]*cloud2.shape[0]/(cloud2.shape[0]*(n_points-1))`
`        return overlap_1_2, overlap_2_1`
`    #Compute k neighbours graph`
`    if distance_metric == 'euclidean':`
`        if use_fast:`
`            index = faiss.IndexFlatL2(cloud_all.shape[1])   # build the index`
`            index.add(cloud_all) # add vectors to the index`
`            _, I = index.search(cloud_all, k+1)`
`            I = I[:,1:]`
`        else:`
`            knn = NearestNeighbors(n_neighbors=k, metric="minkowski", p=2).fit(`
`                cloud_all`
`            )`
`            I = knn.kneighbors(return_distance=False)`
` `            
`    elif distance_metric == 'geodesic':`
`        model_iso = Isomap(n_components = 1)`
`        emb = model_iso.fit_transform(cloud_all)`
`        dist_mat = model_iso.dist_matrix_`
`        knn = NearestNeighbors(n_neighbors=k, metric="precomputed").fit(dist_mat)`
`        I = knn.kneighbors(return_distance=False)`
` `
`    #Compute overlapping`
`        #total fraction of neighbors that belong to the other cloud`
`    overlap_1_2 = np.sum(I[:idx_sep,:]>=idx_sep)/(cloud1.shape[0]*k)`
`    overlap_2_1 = np.sum(I[idx_sep:,:]<idx_sep)/(cloud2.shape[0]*k)`
``
`    return overlap_1_2, overlap_2_1`

With this we avoid any uneeded computations and also apply the case for the geodesic case.
If you wanna implement it in your code and update the pull request I can accept it!
